src/domains/ecommerce-common/03_storage.tf

```diff
@@ -391,17 +384,100 @@ resource "azurerm_monitor_scheduled_query_rules_alert" "ecommerce_transient_enqu
   enabled        = true
   query = format(<<-QUERY
     let OpCountForQueue = (operation: string, queueKey: string) {
-      StorageQueueLogs
-      | where OperationName == operation and ObjectKey startswith queueKey
-      | summarize count()
+        StorageQueueLogs
+        | where OperationName == operation and ObjectKey startswith queueKey
+        | summarize count() 
+        | project count_ 
+        | extend dummy=1
+    };
+    let PutMessages = (queueName: string) {
+      OpCountForQueue("PutMessage", queueName)
+        | project PutCount = count_
+        | extend dummy = 1
+    };
+    let DeletedMessages =  (queueName: string) {
+      OpCountForQueue("DeleteMessage", queueName)
+        | project DeleteCount = count_
+        | extend dummy = 1
     };
     let MessageRateForQueue = (queueKey: string) {
-      OpCountForQueue("PutMessage", queueKey)
-      | join kind=fullouter OpCountForQueue("DeleteMessage", queueKey) on count_
-      | project name = queueKey, Count = count_ - count_1
+        PutMessages(queueKey)
+        | join kind=inner (DeletedMessages(queueKey)) on dummy
+        | extend Diff = PutCount - DeleteCount
+        | project PutCount, DeleteCount, Diff
     };
     MessageRateForQueue("%s")
-    | where Count > ${each.value.threshold}
+    | where Diff > ${each.value.threshold}
+    QUERY
+    , "/${module.ecommerce_storage_transient.name}/${local.project}-${each.value.queue_key}"
+  )
+  severity    = each.value.severity
+  frequency   = each.value.frequency
+  time_window = each.value.time_window
+  trigger {
+    operator  = "GreaterThan"
+    threshold = 0
+  }
+}
+
+locals {
+  queue_expiration_alert_props = var.env_short == "p" ? [
+    {
+      "queue_key"   = "transactions-expiration-queue"
```

Comment:
we should add here also the other queues were events are written with a visibility timeout other than expiration queue.
here should be add also:
- all the \*retry\* queues (such as notifications-service-retry-queue, transaction-notifications-retry-queue and so on)
- transaction-auth-requested-queue (for NPG transactions event is re-written into visibility timeout to trigger GET state api call)
- transaction-auth-outcome-waiting-queue

all the above mentioned queues have events written with visibility timeout that could cause put - delete count to fail for a determined time slot

Reply:
You're right. I'm going to add those queues to this alert group

Reply:
Done here d34f9d6df049659827b7e46df82da0bde826a8f5